lib/blake2.rb

```diff
@@ -1,2 +1,29 @@
 require 'blake2_ext'
 require 'blake2/key'
+
+class Blake2
+  def self.hex(input, key = Blake2::Key.none, out_len = 32)
+    check_if_valid!(input, key, out_len)
+    Blake2.new(out_len, key).digest(input, :to_hex)
+  end
+
+  def self.bytes(input, key = Blake2::Key.none, out_len = 32)
+    check_if_valid!(input, key, out_len)
+    Blake2.new(out_len, key).digest(input, :to_bytes)
+  end
+
+  def self.check_if_valid!(input, key, out_len)
+    unless input.is_a?(String)
+      raise ArgumentError, 'input arg must be a String'
+    end
+
+    unless key.is_a?(Blake2::Key)
+      raise ArgumentError, 'key arg must be a Blake2::Key'
+    end
+
+    unless out_len.is_a?(Integer) && out_len.between?(1, 32)
+      raise ArgumentError, 'out_len arg must be an Integer between 1 and 32 inclusive'
+    end
+  end
+  private_class_method :check_if_valid!
```

Comment:
I had a long thought about this. Here's what we could do: instead of Blake2 being a class, it could be a module. The original Blake2 class would be nested within that module.

The module's responsibility would be to:

1. Act as a namespace 😅 
2. Provide the simple config you're mentioning.

I wasn't in favor of making it easy to use `Key.none`s, but I guess that's a real-world use-case people have.

Also: please use keyword arguments instead of named arguments. Thanks!